bundle/Command/TagContentByTypesCommand.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\SiteBundle\Command;
+
+use Ibexa\Contracts\Core\Repository\Repository;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Ibexa\Contracts\Core\Repository\Values\Content\Query;
+use Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion;
+use Ibexa\Contracts\Core\SiteAccess\ConfigResolverInterface;
+use Netgen\TagsBundle\API\Repository\TagsService;
+use Netgen\TagsBundle\API\Repository\Values\Tags\Tag;
+use Netgen\TagsBundle\Core\FieldType\Tags\Value as TagFieldValue;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Exception\InvalidOptionException;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Throwable;
+
+use function array_filter;
+use function array_map;
+use function array_unique;
+use function array_values;
+use function explode;
+use function sprintf;
+use function trim;
+
+final class TagContentByTypesCommand extends Command
+{
+    private SymfonyStyle $style;
+    private InputInterface $input;
+
+    public function __construct(
+        private Repository              $repository,
+        private TagsService             $tagsService,
+        private ConfigResolverInterface $configResolver,
+    ) {
+        // Parent constructor call is mandatory for commands registered as services
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription('Add tags to content')
+            ->addOption('parent-location', null, InputOption::VALUE_REQUIRED)
+            ->addOption('content-types', null, InputOption::VALUE_REQUIRED)
+            ->addOption('tag-id', null, InputOption::VALUE_REQUIRED)
+            ->addOption('field-identifier', null, InputOption::VALUE_OPTIONAL);
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->input = $input;
```

Comment:
You don't need the `$this->input` property. You already have it provided in `execute` method arguments. And as for other methods, just transfer the value or the option to them.

Reply:
solved in 3dd233e7